dprof.js

```diff
@@ -110,39 +114,42 @@ asyncHook.addHooks({
   destroy: asyncDestroy
 });
 
-const root = new Node({ constructor: { name: 'root' } }, getCallSites(2));
-      root.rootIntialize();
+const root = new Node(
+  0, { 'constructor': { name: 'root' } },
+  getCallSites(2),
+  null
+);
+root.rootIntialize();
 
-const stateMap = new Map();
+const nodes = new Map();
 let currState = root;
 
 function asyncInit(uid, handle, provider, parentUid) {
   // get parent state
-  const state = (parentUid === null ? currState : stateMap.get(parentUid));
+  const state = (parentUid === null ? currState : nodes.get(parentUid));
 
   // add new state node
-  stateMap.set(uid, state.add(handle));
+  nodes.set(uid, state.add(uid, handle));
 }
 
 function asyncBefore(uid) {
-  const state = stateMap.get(uid);
+  const state = nodes.get(uid);
 
   state.before();
   currState = state;
 }
 
 function asyncAfter(uid) {
-  const state = stateMap.get(uid);
+  const state = nodes.get(uid);
 
   state.after();
   currState = root;
 }
 
 function asyncDestroy(uid) {
-  const state = stateMap.get(uid);
+  const state = nodes.get(uid);
 
   state.destroy();
-  stateMap.delete(uid);
```

Comment:
Not more than before. Before all node objects where stored in a tree structure (`.children`), now they are also stored in the `nodes` Map. So yes it is _leaky_, but all the information needs to be used in the end when the tree is saved into `dprof.json.gz`.

I guess we could write it in chunks to `dprof.json` now that it's a flat data structure.